CLIENT_GENERATOR_GUIDE.md

```diff
@@ -0,0 +1,62 @@
+# Generate Clients for OpenSearch using OpenAPI Specification
+
+OpenSearch provides clients for OpenSearch in multiple languages. The biggest challenge with this is keeping the clients up to date with the latest changes in OpenSearch. To solve this problem, we're automating the process of generating clients for OpenSearch using the OpenAPI specification. While OpenAPI comes with many well established off-the-shelf client generators for most languages, the OpenSearch APIs come with a lot of quirkiness that makes it near impossible to use these off-the-shelf generators. For this reason, we've opted to write our own client generators that is specifically tailored to the OpenSearch APIs. This document will walk you through the process of generating clients, more specifically client API methods.
+
+## The Grouping of API Operations
+The OpenSearch clients, though written in different languages for different frameworks, share one distinct characteristic: API Operations are grouped into actions and namespaces. Operations serving the same functionality are often grouped together into a single API Action, which is represented in the client as an API method.
+
+The **search** action, for example, is consisted of 4 operations:
+- `GET /_search`
+- `POST /_search`
+- `GET /{index}/_search`
+- `POST /{index}/_search`
+
+The **indices.get_field_mapping**, is consisted of 2 operations:
+- `GET /_mapping/field/{fields}`
+- `GET /{index}/_mapping/field/{fields}`
+
+In a client, the `search` operations are grouped in to a single API method, `client.search(...)`, and the `indices.get_field_mapping` operations are grouped into a single API method, `get_field_mapping` of the `indices` namespace, `client.indices.get_field_mapping(...)`
+
+In the OpenAPI spec, this grouping is denoted by `x-operation-group` vendor extension in every operation definition. The value of this extension is the name of the API action (like `search` or `indices.get_field_mapping`). Operations with the same `x-operation-group` value are guaranteed to have the same query string parameters, response body, and request body (for PUT/POST/DELETE operations). Common path parameters are also guaranteed to be the same. The only differences between operations are the HTTP method and the path. With that in mind, below are rules on how to combine operations of different HTTP methods and path compositions.
+
+- If a deprecated operation is present in the group AND there are other non-deprecated operations in the group, then ignore the deprecated operation.
```

Comment:
Yes. Can't reference them yet because the current commits don't have the example operations in the `Opensearch.openapi.json` file yet. Will add it in a future PR.